train.py

```diff
@@ -0,0 +1,117 @@
+import os
+import time
+
+import hydra
+import torch
+import psutil
+import imageio
+import numpy as np
+from omegaconf import DictConfig
+from torchvision.utils import save_image
+
+import wandb
+from utils import get_interpolations
+from models.autoencoder import Autoencoder
+from train_utils import test_epoch, train_epoch
+
+
+def initialize_model(args: DictConfig):
+    if args.model.type == "AE":
+        model = Autoencoder(args)
+    else:
+        raise NotImplementedError("Model type not implemented")
+    return model
+
+
+def train(args: DictConfig, model: torch.nn.Module):
```

Comment:
```suggestion
def train(args: DictConfig, autoencoder: Autoencoder):
```

I think you're not actually using this function with an instance of `nn.Module`, but of `Autoencoder`. We're not using [mypy](https://mypy-lang.org/), but you can take a look at what it does, as it would probably complain here as well.

It's is great to be consistent with the variable naming across multiple functions, so if something is called "model" in `main` it should also be called "model" in `train`. Right now there's a mix.